node/tests/testcase.go

```diff
@@ -197,8 +198,8 @@ func (tcase testcase) Test(t *testing.T, np, npr datamodel.NodePrototype) {
 		if n.(schema.TypedNode).Representation().Kind() == datamodel.Kind_Map {
 			t.Run("repr-create with AK+AV", func(t *testing.T) {
 				n3, err := shallowCopyMap(npr, n.(schema.TypedNode).Representation())
-				Wish(t, err, ShouldEqual, nil)
-				Wish(t, datamodel.DeepEqual(n, n3), ShouldEqual, true)
+				qt.Check(t, err, qt.Equals, nil)
+				qt.Check(t, datamodel.DeepEqual(n, n3), qt.Equals, true)
```

Comment:
mental TODO: in the future, we might be able to expose datamodel.DeepEqual as a go-cmp Comparer. Definitely out of scope for this refactor, though.

Reply:
One could also use the `printer` package and then do string diff on the results.

That would simultaneously be very complete, and produce very readable debugable output, and also happily ignore the actual node implementation strategy (which is usually what we want).

Reply:
`quicktest` comes with [`CmpEquals`](https://pkg.go.dev/github.com/frankban/quicktest#hdr-CmpEquals) which looks pretty flexible; I wonder if we could use that here?

Reply:
That is what I was alluding to. But you need a custom comparer, because by default, CmpEquals will just loop infinitely. This is because bindnode ipld.Node implementations contain schema types, which have cyclic pointers.

Reply:
I originally arrived there via the flat datamodel.DeepEqual, which works, but also doesn't give good errors. I think either a custom go-cmp comparer, or comparing encoded versions like "printer" or "dagjson" like Eric mentions, would work better.

Reply:
I attempted to write a checker that works with `printer` [here](https://github.com/ipld/go-ipld-prime/pull/291/commits/3c062db77fe81b0db0237f419be5b52ac9316005) but looks like there are TODOs in the printer that tests hit. Going to use `datamodel.DeepEqual` going forward.